homeassistant/components/hdmi_cec.py

```diff
@@ -4,118 +4,535 @@
+            type('call', (object,), {'data': {ATTR_DST: self._logical_address, ATTR_CMD: 0x8C}}))
+
+    @asyncio.coroutine
+    def async_request_cec_osd_name(self):
+        self.cec_client.tx(
+            type('call', (object,), {'data': {ATTR_DST: self._logical_address, ATTR_CMD: 0x46}}))
+
+    @asyncio.coroutine
+    def async_request_cec_power_status(self):
+        self.cec_client.tx(
+            type('call', (object,), {'data': {ATTR_DST: self._logical_address, ATTR_CMD: 0x8F}}))
+
+    @callback
+    def _update_callback(self, event):
+        if ATTR_CMD not in event.data or ATTR_SRC not in event.data:
+            return
+        cmd = event.data[ATTR_CMD]
+        src = event.data[ATTR_SRC]
+        dst = event.data[ATTR_DST] if ATTR_DST in event.data else None
+        cmd_chain = event.data[ATTR_ATT] if ATTR_ATT in event.data else []
+        if src == self._logical_address:
+            if cmd == 0x90:
+                status = cmd_chain[0]
+                if status == 0x00:
+                    self._state = STATE_ON
+                elif status == 0x01:
+                    self._state = STATE_OFF
+                else:
+                    self._state = STATE_UNKNOWN
+                _LOGGER.info("Got status for device %x -> %x, %s", src, dst, self._state)
+            elif cmd == 0x47:
+                self._name = ''
+                for c in cmd_chain:
+                    self._name += chr(c)
+                _LOGGER.info("Got name for device %x -> %x, %s", src, dst, self._name)
+            elif cmd == 0x87:
+                self._vendor_id = 0
+                for i in cmd_chain:
+                    self._vendor_id *= 0x100
+                    self._vendor_id += i
+                _LOGGER.info("Got vendor for device %x -> %x, %s: %s", src, dst, self.vendor_id, self.vendor_name)
+            elif cmd == 0x84:
+                self._physical_address = "%d.%d.%d.%d" % (
+                    cmd_chain[0] / 0x10, cmd_chain[0] % 0x10, cmd_chain[1] / 0x10, cmd_chain[1] % 0x10)
+                self._cec_type_id = cmd_chain[2]
+                _LOGGER.info("Got physical address and type for device %x -> %x, %s, %s",
+                             src, dst, self._physical_address, self.type)
+
+            self.schedule_update_ha_state(False)
+
+    @property
+    def is_on(self):
+        """Return true if device is on."""
+        return self._state
+
+    @property
+    def name(self):
+        """Return the name of the device."""
+        return "%s %s" % (self.vendor_name, self._name) if self._name is not None and self.vendor_name is not None \
+                                                           and self.vendor_name != 'Unknown' \
+            else "%s %d" % (DEVICE_TYPE_NAMES[self._cec_type_id], self._logical_address) if self._name is None \
+            else "%s %d (%s)" % (
+            DEVICE_TYPE_NAMES[self._cec_type_id], self._logical_address, self._name)
+
+    @property
+    def state(self):
```

Comment:
You cannot overwrite the state as this is a generic entity. State property is owned by the abstract base classes of the components (like media player)